src/common/lib/client/realtimepresence.ts

```diff
@@ -435,7 +431,7 @@ class RealtimePresence extends EventEmitter {
   _synthesizeLeaves(items: PresenceMessage[]): void {
     const subscriptions = this.subscriptions;
     items.forEach(function (item) {
-      const presence = presenceMessageFromValues({
+      const presence = PresenceMessage.fromValues({
```

Comment:
I think we switched to standalone functions rather than static methods because that makes them tree-shakable separately from the class, so for example if someone is using the `PresenceMessage` class but isn't using `presenceMessageFromValues` then the standalone function isn't imported. I guess this would only happen if someone is using rest presence, since we use `fromValues` internally here so anyone using realtime presence will be using it either way.

Reply:
fwiw i think the minified size of that function is probably so small that it doesn't really matter anyway but just fyi

Reply:
Thanks, but.. yeah, I'm very disinclined to spend any brainpower worrying about that one-line function now being tree-shaken in that specific circumstance

fwiw it looks like the minimal bundle size after this pr is now down about 3/4 of a kilobyte, mostly from improved BaseMessage codesharing and removing some unnecessary decoding abstractions in realtimechannel. I'd argue that worrying overly much about making sure these sort of scraps are tree-shakable while ignoring the low-hanging fruit (neglected unnecessarily long code in non-tree-shakable parts of the lib) is misdirected energy